### ISSUE TITLE
Atom hangs when "}" is deleted when editing .thrift file

### ISSUE THREAD

**Issue by prashantv** [atom/atom#7644]
I have the language-thrift package (https://atom.io/packages/language-thrift) installed.

When editing a Thrift file, enter the following code:
```thrift
exception SimpleErr {
  1: string message
}


service SimpleService {
  void Simple() throws (1: SimpleErr simpleErr)
}
```

If you delete the "}" at the end of the exception block, the whole editor freezes up.

This seems to be an issue with the grammar / syntax parsing.

**Comment by kubetz** [comment]
If you close all running instances, run atom in safe mode with `atom --safe` and it stops freezing, then it is not a problem of atom/atom you need to open an issue for [language-thrift plug-in here](https://github.com/davidcelis/language-thrift/issues).

**Comment by izuzak** [comment]
I was able to reproduce this, but not sure where the problem might be. @nathansobo can you take a quick look? Could a "bad" grammar cause such behavior and is there anything we can do about that?

**Comment by nathansobo** [comment]
Taking a look.

**Comment by nathansobo** [comment]
The fix for this is nontrivial, but I'm working on it. Could take some time.

**Comment by nathansobo** [comment]
Thanks for the detailed report and sorry for the hassle. This will be fixed in 1.0.1, scheduled to go out this week.

**Comment by prashantv** [comment]
Thanks for fixing this so quickly Nathan! Looking forward to the next release
